src/main/java/com/beust/jcommander/MissingCommandException.java

```diff
@@ -26,11 +26,23 @@
 @SuppressWarnings("serial")
 public class MissingCommandException extends ParameterException {
 
-  public MissingCommandException(String string) {
-    super(string);
+  /**
+   * the command passed by the user.
+   */
+  private final String unknownCommand;
+
+  public MissingCommandException(String message) {
+    super(message);
+    this.unknownCommand = null;
+  }
+
+  public MissingCommandException(String command, String message) {
+    super(message);
+    this.unknownCommand = command;
```

Comment:
You are right, keeping the order of arguments in the constructor makes it much better.
For the use of `this`, I am worried: it won't compile unless I remove the `final` modifier on the `unknownCommand` field.

Which one do you prefer: having a non final `unknownCommand` field, or using `this` constructor ?
I agree the use of `this` can help reduce the likeliness of introducing a bug later on, but having the field `final` also ensures no further constructor will forget to handle this field too.